### PR TITLE
Log with :info level instead of :warn when AppSignal is disabled

### LINKED ISSUE #335: Disable appsignal printout in tests
I already have this in my `config/test.exs` (and it is being loaded in the test environment)

```
config :appsignal, :config,
  active: false
```

Yet I still have this line showing up in my test output when running `mix test`

    09:36:56.086 [warn]  Warning: No valid AppSignal configuration found, continuing with AppSignal metrics disabled.

Is there anything I can do to suppress that line?

### PR DESCRIPTION
**Pull request by jeffkreeftmeijer**
Closes #335. Thanks for reporting this, @axelson!

If AppSignal is not active, we'll :info-log the "AppSignal disabled"
line. This will show in a Phoenix app in development, but not in tests
(as the default :logger level for tests is :warn in Phoenix).

We'll only warn if the configuration is invalid, but AppSignal was set
to active (like when omitting the push_api_key from
config/appsignal.exs).

### CODE REVIEW

File: lib/appsignal/config.ex [hunk 1]
```diff
@@ -56,10 +56,7 @@ defmodule Appsignal.Config do
   Returns whether the AppSignal agent is configured to start on application launch.
   """
   @spec active?() :: boolean
-  def active? do
-    config = Application.fetch_env!(:appsignal, :config)
-    config.valid && config.active
-  end
+  def active?, do: Application.fetch_env!(:appsignal, :config).active
```
Comment (tombruijn) [inline lib/appsignal/config.ex]: This implementation differs from the Ruby gem and I'm not sure we should change that.

https://github.com/appsignal/appsignal-ruby/blob/4ad818d75f113530817ee14af3b3e2af4052b728/lib/appsignal/config.rb#L140-L142

The config can only be active if the config is valid.

**Comment by matsimitsu** [comment]
Support-wise what is the default logger level for production apps? Ideally it will still show up there, otherwise we have nothing to go on when the integration is not starting.

File: lib/appsignal/config.ex (hunk 1 above)
Comment (jeffkreeftmeijer) [reply lib/appsignal/config.ex]: Exactly. This is the only place `Appsignal.Config.active?/0` is used (which now corresponds to the value in the `:active` configuration and the `APPSIGNAL_ACTIVE` environment variable).

https://github.com/appsignal/appsignal-elixir/blob/7af417d705f7935358ed1f500995ea9ca833a046/lib/appsignal.ex#L65-L89

`Config.initialize()` will return `{:error, :invalid_config}` when the configuration is invalid, disabling the integration. 

I could rename this function, but I need a way to determine if the integration was configured to be active here.

**Comment by jeffkreeftmeijer** [comment]
@matsimitsu it will. The default log level is `:info`. Phoenix prepares a `test/config.exs` which bumps the level to warn. This will hide the `Logger.info("AppSignal disabled.")`, but it will show `Logger.warn("Warning: No valid AppSignal configuration found, continuing with " <> "AppSignal metrics disabled.")`, if the integration is somehow activated in the test env.

File: lib/appsignal/config.ex (hunk 1 above)
Comment (jeffkreeftmeijer) [reply lib/appsignal/config.ex]: Will rename (`should_be_active?/0`) and keep `active?/0` as is.

Comment (tombruijn) [reply lib/appsignal/config.ex]: Moving PR back to WIP until the changes are made.

**Pull request merged**